plugins/woocommerce/includes/gateways/paypal/class-wc-gateway-paypal.php

```diff
@@ -346,6 +348,22 @@ public function admin_options() {
 	 */
 	public function init_form_fields() {
 		$this->form_fields = include __DIR__ . '/includes/settings-paypal.php';
+		$this->form_fields = apply_filters( 'woocommerce_settings_api_form_fields_paypal', $this->form_fields );
+	}
+
+	/**
+	 * Filter to remove fields for Orders v2.
+	 *
+	 * @param array $form_fields Form fields.
+	 * @return array
+	 */
+	public function maybe_remove_fields( $form_fields ) {
+		// Additional details are added to the receiver email when subscriptions are enabled.
+		// We don't need this for Orders v2.
+		if ( WC_Gateway_Paypal_Helper::should_use_orders_v2() ) {
+			unset( $form_fields['receiver_email'] );
+		}
+		return $form_fields;
```

Comment:
Subscription injects additional description to the `receiver_email` field. Without this function, there will be a floating email field and description, though we have hidden it in the settings file. It's because the filter in subscriptions runs later than `init_form_fields`.

Reply:
![Uploading Screenshot 2025-07-29 at 3.15.22 PM.png…]()